src/libraries/System.Private.CoreLib/src/System/IndexOfAnyValues/IndexOfAnyValues.cs

```diff
@@ -0,0 +1,283 @@
+
+                return Sse3.IsSupported && lookup.Contains(0)
+                    ? new IndexOfAnyAsciiCharValues<IndexOfAnyAsciiSearcher.Ssse3HandleZeroInNeedle>(bitmap, lookup)
+                    : new IndexOfAnyAsciiCharValues<IndexOfAnyAsciiSearcher.Default>(bitmap, lookup);
+            }
+
+            if (values.Length <= 5)
+            {
+                Debug.Assert(values.Length is 3 or 4 or 5);
+                return values.Length switch
+                {
+                    3 => new IndexOfAny3Values<char, short>(shortValues),
+                    4 => new IndexOfAny4Values<char, short>(shortValues),
+                    _ => new IndexOfAny5Values<char, short>(shortValues),
+                };
+            }
+
+            if (maxInclusive < 256)
+            {
+                // This will also match ASCII values when IndexOfAnyAsciiSearcher is not supported
+                return new IndexOfAnyLatin1CharValues(values);
+            }
+
+            return values.Length < Vector128<short>.Count
+                ? new IndexOfAnyCharValuesProbabilistic<ShortLoopContains>(values)
+                : new IndexOfAnyCharValuesProbabilistic<StringContains>(values);
+        }
+
+        private static bool TryGetSingleRange<T>(ReadOnlySpan<T> values, out T maxInclusive, [NotNullWhen(true)] out IndexOfAnyValues<T>? singleRangeValues)
+            where T : struct, INumber<T>, IMinMaxValue<T>
+        {
+            singleRangeValues = null;
+
+            T min = T.MaxValue;
+            T max = T.MinValue;
+
+            foreach (T value in values)
+            {
+                min = T.Min(min, value);
+                max = T.Max(max, value);
+            }
+
+            maxInclusive = max;
+
+            uint range = uint.CreateChecked(max - min) + 1;
+            if (range > values.Length)
+            {
+                return false;
+            }
+
+            Span<bool> seenValues = range < 256 ? stackalloc bool[256] : new bool[range];
+            seenValues = seenValues.Slice(0, (int)range);
+            seenValues.Clear();
+
+            foreach (T value in values)
+            {
+                int offset = int.CreateChecked(value - min);
+                seenValues[offset] = true;
+            }
+
+            if (seenValues.Contains(false))
+            {
+                return false;
+            }
+
+            singleRangeValues = (IndexOfAnyValues<T>)(object)new IndexOfAnyValuesInRange<T>(min, max);
+            return true;
+        }
+
+        internal interface IAsciiSet
+        {
+            public static abstract Vector128<byte> Bitmap { get; }
+            public static abstract bool Contains(char c);
+        }
+
+        private readonly struct AsciiLetter : IAsciiSet
+        {
+            public static Vector128<byte> Bitmap => Vector128.Create(160, 240, 240, 240, 240, 240, 240, 240, 240, 240, 240, 80, 80, 80, 80, 80);
```

Comment:
Can you add a comment that the values are pre-computes based on the used algorithm, or similar.
Now (and for us) it's clear where these values come from, but it's not obvious.